src/qwircl/core.cljs

```diff
@@ -134,19 +226,79 @@
           (draw-empty-space x y (:highlighted? cell)))
         (draw-empty-space x y)))))
 
-(defn draw-header [player]
+(defn button-action [button pda]
+  (condp = button
+    :undo :undo
+    :submit (condp = (:s pda)
+              :picking (if (empty? (:positions pda)) 
+                         :trade
+                         :submit)
+              :playing :submit
+              :submit)
+    :submit))
+
+(defn button-status [button pda]
+  (condp = (:s pda)
+    :initial :inactive 
+    :playing :active
+    :picking (condp = button 
+               :submit :active
+               :undo :active
+               :inactive)))
+
+(defn draw-button [button x pda]
+  (let [status (button-status button pda)]
+    (set-color (condp = status
```

Comment:
Some of these can be implemented with map literals: `({:inactive :background :active :light-green} status)`

Reply:
True. I left the condp in case I needed something more complicated.